h/static/scripts/directive/annotation.js

```diff
@@ -436,6 +442,10 @@ function AnnotationController(
     */
   function view() {
     vm.action = 'view';
+
+    // trigger an immediate update of the timestamp when switching
+    // back to the view mode just after creating or updating an annotation
+    updateTimestamp();
```

Comment:
Should we call this in `updateViewModel()`, rather than here? `updateViewModel()` is called both after saving a new annotation and after updating an annotation. `updateViewModel()` is also called on `init()`, so you could remove the call to `updateTimestamp()` in `init()` as well. The idea for `updateViewModel()` was that it would become the _only_ function that updates any of the values on `vm`, and it would be called at just the right times.